src/braket/experimental/autoqasm/types/types.py

```diff
@@ -81,16 +85,19 @@ def __init__(self, *args, **kwargs):
 class BoolVar(oqpy.BoolVar):
     def __init__(self, *args, **kwargs):
         super(BoolVar, self).__init__(*args, **kwargs)
-        self.name = program.get_program_conversion_context().next_var_name(oqpy.BoolVar)
+        if "name" not in kwargs:
+            self.name = program.get_program_conversion_context().next_var_name(oqpy.BoolVar)
 
 
 class FloatVar(oqpy.FloatVar):
     def __init__(self, *args, **kwargs):
         super(FloatVar, self).__init__(*args, **kwargs)
-        self.name = program.get_program_conversion_context().next_var_name(oqpy.FloatVar)
+        if "name" not in kwargs:
```

Comment:
if I do `aq.FloatVar(name="name", val)`, I don't want that name to be replaced

Reply:
Are you using this functionality somewhere (other than in the tests)? Just curious why we would need it - I think internally we have been using `oqpy.FloatVar` directly when we need to specify a name, and the AutoQASM user scenario for this is to assign to the desired name, e.g. `my_var_name = aq.FloatVar(val)` and the assignments operator takes care of naming the oqpy variable correctly.

Reply:
I had at one point, I think I'm not anymore. I still think we should have this, but I could move it into a separate PR if you want.

One reason we'd want it is that we are trying to move away from calling oqpy directly anywhere we want. I also feel like it's incorrect to accept a kwarg (well, oqpy accepts it, and the object has such an attribute) and then overwrite it.

Reply:
Yeah, makes sense. I guess ultimately these `aq.*Var` types will probably not derive from the `oqpy` types anymore, so this logic might change again at that point.

If we do support `name` here, would it make sense to add it explicitly as an (optional) parameter to `__init__`? Otherwise someone would have to look at the implementation to know that it exists. Same goes for `val`, actually - if we want to truly abstract the oqpy API away, we should have an explicit signature for `__init__` here rather than just passing all the args and kwargs to oqpy.

And yes since it's independent, moving this part to a separate PR might make sense, if that's not too much overhead for you.

Reply:
Yes I'll move it 👍

Reply:
Working on a draft now: https://github.com/amazon-braket/amazon-braket-sdk-python/pull/793/files